compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3216,6 +3215,252 @@ TR::Register *OMR::X86::TreeEvaluator::arraycopyEvaluator(TR::Node *node, TR::Co
    return NULL;
    }
 
+static void generateRepMovsInstruction(TR_X86OpCodes REPMOVS, TR::Node *node, TR::Register* sizeRegister, TR::RegisterDependencyConditions* dependencies, TR::CodeGenerator *cg)
+   {
+   switch (REPMOVS)
+      {
+      case REPMOVSQ:
+         generateRegImmInstruction(SHRRegImm1(), node, sizeRegister, 3, cg);
+         break;
+      case REPMOVSD:
+         generateRegImmInstruction(SHRRegImm1(), node, sizeRegister, 2, cg);
+         break;
+      case REPMOVSW:
+         generateRegImmInstruction(SHRRegImm1(), node, sizeRegister, 1, cg);
```

Comment:
Good point. I would definitely do if the instruction choices are kept the same. I optimized generated code from this functions to generate shorter instructions; now the instruction formats are different; and hence I have to keep this switch table.